src/components/forms/JurisdictionSelect/index.tsx

```diff
@@ -46,14 +46,39 @@ export interface JurisdictionSelectProps<T = SelectOption> extends AsyncSelectPr
 export const promiseOptions = (
   service: OpenSRPService,
   paramsToUse: Dictionary,
-  hierarchy: SelectOption[]
+  hierarchy: SelectOption[],
+  jurisdictionStatus: boolean,
+  setFinalLocation: (value: boolean) => void,
+  setJurisdictionParam: (value: boolean) => void
 ) =>
   // tslint:disable-next-line:no-inferred-empty-object-type
-  new Promise((resolve, reject) =>
+  new Promise((resolve, reject) => {
     service
-      .list(paramsToUse)
-      .then((e: JurisdictionOption[]) => {
-        const options = e.map(item => {
+      .list({ ...paramsToUse, is_jurisdiction: jurisdictionStatus })
+      .then((jurisdictionLocationOptions: JurisdictionOption[]) => {
+        /** Check if payload has no name property then use id instead
+         *  If there is no location return no options
+         */
+        if (!jurisdictionStatus) {
+          setFinalLocation(true);
+          if (
+            jurisdictionLocationOptions.length >= 1 &&
+            !jurisdictionStatus &&
+            !jurisdictionLocationOptions[0].properties.name
```

Comment:
Why do we need to check for just the first one in the array?  Whatever we do next we need apply to every item in the array.  And keep in mind that we can have multiple items returned.